### PR TITLE
initialize api handlers by APIBackend

### PR DESCRIPTION
**Pull request by zhulongcheng**
Related #2140

  - [ ] CHANGELOG.md updated with a link to the PR (not the Issue)
  - [x] Rebased/mergeable
  - [x] Tests pass
  - [ ] swagger.json updated (if modified Go structs or API)
  - [x] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)

### CODE REVIEW

File: http/api_handler_test.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+		UserResourceMappingService: mock.NewUserResourceMappingService(),
+		LabelService:               mock.NewLabelService(),
+		DashboardService:           &mock.DashboardService{},
+		ViewService:                &mock.ViewService{},
+		MacroService:               &mock.MacroService{},
+		BasicAuthService:           mock.NewBasicAuthService("", ""),
+		TaskService:                &mock.TaskService{},
+		SecretService:              mock.NewSecretService(),
+
+		// TODO impl mock service
```
Comment (desa) [inline http/api_handler_test.go]: Could you create an issue for these?

Comment (zhulongcheng) [reply http/api_handler_test.go]: Of course.   #2145

**Comment by zhulongcheng** [comment]
Rebased.

**Comment by desa** [comment]
@zhulongcheng I really like this change, but when I started to implement it for some of the closed source pieces I hit a few panics. What do you think about having custom backends for each of the handlers. This way we know exactly which handlers require which services.

(Sorry for the early approval)

**Comment by zhulongcheng** [comment]
@desa I think custom-backend is a better way to know that which services required by a handler.

What is the detail of those panics? custom-backend is an effective way to avoid/reduce panics?

**Comment by desa** [comment]
👍 

> What is the detail of those panics? custom-backend is an effective way to avoid/reduce panics?

I could chase down each of the panics, but they were the result of the service not being set on the `APIBackend` and to figure out which things needed to be set I had to look at the constructor. I could have gone through and fixed the specific panic, but I think it might be easier if each handler had its own specific backend for the constructor. What do you think?

**Pull request closed without merging**